From d339c9d64fa9a15ff6d933affd3902e33263b4ed Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Fri, 4 Apr 2014 17:28:51 +0800
Subject: [PATCH] net: macb: workaround for macb1

As the MACB1 share pin with LCD, so when design hardware, there
is an buffer chip bwteen these two devices, which cause the phy
can not get the reset strap value correctly.

Add a workaround to make the macb1 working.

Signed-off-by: Bo Shen <voice.shen@atmel.com>
Acked-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 drivers/net/macb.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/net/macb.c b/drivers/net/macb.c
index 79334bb..1d21f63 100644
--- a/drivers/net/macb.c
+++ b/drivers/net/macb.c
@@ -766,6 +766,18 @@ int macb_eth_initialize(int id, void *regs, unsigned int phy_addr)
 	miiphy_register(netdev->name, macb_miiphy_read, macb_miiphy_write);
 	macb->bus = miiphy_get_dev_by_name(netdev->name);
 #endif
+
+#ifdef CONFIG_MACB1
+	/*
+	 * As the MACB1 share pin with LCD, so when design hardware, there
+	 * is an buffer chip bwteen these two devices, which cause the phy
+	 * can not get the reset strap value correctly.
+	 * Add this workaround to make macb1 working.
+	 */
+	if (id == 1)
+		macb_mdio_write(macb, 0x16, 0x2);
+#endif
+
 	return 0;
 }
 
-- 
1.9.1

